extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -81,8 +88,34 @@ public List<InterceptorBinding> getAdditionalBindings() {
     }
 
     @BuildStep
-    void transformWithSpan(
-            BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+    void transformWithSpan(BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+
+        // Transform deprecated annotation into new one
+        annotationsTransformer.produce((new AnnotationsTransformerBuildItem(context -> {
+
+            AnnotationTarget target = context.getTarget();
+
+            if (target.kind().equals(AnnotationTarget.Kind.METHOD)) {
+                Set<AnnotationInstance> legacyWithSpans = context.getAnnotations().stream()
+                        .filter(annotationInstance -> annotationInstance.name()
+                                .equals(DotName.createSimple(io.opentelemetry.extension.annotations.WithSpan.class.getName())))
```

Comment:
It is common to have a `DotNames` class (sitting next to the processor class) holding constants like this.

Reply:
will create constants